orchestrator-salt/src/main/resources/salt-common/salt/monitoring/textfiles/cipa.sh

```diff
@@ -0,0 +1,118 @@
+  # if the first column empty: skip
+  # if the first column is replication_status write only the status as a metrics
+  # if the first column is not replication_status, write a status only metric, and an ipaserver specific value based metric
+  # in case of a metric value is True/False -> transform metric to 0 / 1
+  # in case of a metric value is not a number and not a True/False -> metric value will be 0, but cipa_info label will be filled as the value
+  local line=$1
+  local state_index=$2
+  IFS=' ' read -r -a server_names <<< "$3"
+  IFS=',' read -ra ROW_ARR <<< "$line"
+  if [[ "${ROW_ARR[0]}" == "" ]]; then
+    continue
+  elif [[ "${ROW_ARR[0]}" == "replication_status" ]];then
+    replication_status_metric_value=1
+    if [[ "${ROW_ARR[$state_index]}" == "OK" ]]; then
+      replication_status_metric_value=0
+    fi
+    echo "cipa_${ROW_ARR[0]}{cipa_status=\"${ROW_ARR[$state_index]}\"} $replication_status_metric_value" >> $prom_tmp_file
+  else
+    metric_name="cipa_${ROW_ARR[0]}"
+    status_metric_name="cipa_${ROW_ARR[0]}_status"
+    status_metric_value=1
+    if [[ "${ROW_ARR[$state_index]}" == "OK" ]]; then
+      status_metric_value=0
+    fi
+    echo "$status_metric_name{cipa_status=\"${ROW_ARR[$state_index]}\"} $status_metric_value" >> $prom_tmp_file
+    vals_idx=0
+    for col in ${ROW_ARR[@]} ; do
+      if [[ "$vals_idx" != "0" && "$vals_idx" != "$state_index" ]]; then
+        srv_idx_for_metric=$(($vals_idx - 1))
+        server_name=${server_names[$srv_idx_for_metric]}
+        if [ -z "${col##[0-9]*}" ]; then
+          echo "$metric_name{cipa_server=\"$server_name\"} $col" >> $prom_tmp_file
+        elif [[ "$col" == "True" ]]; then
+          echo "$metric_name{cipa_server=\"$server_name\"} 0" >> $prom_tmp_file
+        elif [[  "$col" == "False" ]]; then
+          echo "$metric_name{cipa_server=\"$server_name\"} 1" >> $prom_tmp_file
+        else
+          echo "$metric_name{cipa_server=\"$server_name\", cipa_info=\"$col\"} 0" >> $prom_tmp_file
+        fi
+      fi
+      vals_idx=$((vals_idx+1))
+    done
+  fi
+}
+
+prom_file="/var/lib/node_exporter/files/cipa.prom"
+prom_tmp_file="${prom_file}.$$"
+
+if [[ ! -f /var/run/salt-master.pid || ! -f /usr/local/bin/cipa ]]; then
+  echo "[SKIP] Command only runs on salt-master (running) & cipa tool is required."
+  exit 0;
+fi
+
+ADMIN_USER=$(sed '1d' /srv/pillar/freeipa/init.sls | jq -r '.freeipa.admin_user')
+FPW=$(sed '1d' /srv/pillar/freeipa/init.sls | jq -r '.freeipa.password')
+BIND_DN="cn=Directory Manager"
+kinit ${ADMIN_USER} <<< ${FPW} > /dev/null
+DNS_DOMAIN=$(sed '1d' /srv/pillar/freeipa/init.sls | jq -r '.freeipa.domain')
+declare -A IPA_MASTERS=$(sudo ipa config-show | awk -F "[:]" '/IPA masters:/ {print $NF}')
+if [[ ! -d ${HOME}/.config ]]; then mkdir ${HOME}/.config; fi
+cat > /root/.config/checkipaconsistency <<EOF
+[IPA]
+domain = ${DNS_DOMAIN}
+hosts = ${IPA_MASTERS[*]}
```

Comment:
I am thinking we need to also recreate this file every time as it may contain the old IPA_MASTERS before a repair or something like that. It may be best to have a script that makes this file as it will also be used in other places (e.g. diagnostics).

Reply:
i would not include that in this PR but yes, we should create script for that with the diagnostics use case

Reply:
actually this config could be created easily from salt on freeipa instances